src/KubernetesClient/Cache/Store.cs

```diff
@@ -0,0 +1,88 @@
+using System;
+using k8s.Models;
+using System.Collections.Generic;
+using System.Reflection;
+
+namespace k8s.cache
+{
+    public class Store : IStore
+    {
+        private IThreadSafeStore _threadSafeStorage;
+        private Func<IKubernetesObject, string> _metaNamespaceFunc;
+        public string MetaNamespaceKeyFunc(IKubernetesObject o) 
+        {
+            try {
+                 var typedObject = (V1ObjectMeta)o.GetType().GetProperty("Metadata").GetValue(o, null);  
+                if (typedObject.NamespaceProperty == "") {
+                    return typedObject.Name;                
+                }
+                return typedObject.NamespaceProperty+"/"+typedObject.Name;
+            } catch (Exception ex) {
+                Console.WriteLine(ex);
+                return "";
+            }            
+        }
+
+        public Store() {
+            _threadSafeStorage = new ThreadSafeStore();
+            _metaNamespaceFunc = MetaNamespaceKeyFunc;
+        }
+        public void Add(IKubernetesObject o)
+        {            
+            var key = _metaNamespaceFunc(o);            
+            _threadSafeStorage.Add(key, o);            
+        }
+
+        public Tuple<IKubernetesObject, bool> Get(IKubernetesObject o)
```

Comment:
Is a `Tuple` really needed here? I know that's what Go does, but it seems like null and non-null are sufficient in this code.